hudi-client/src/main/java/org/apache/hudi/index/bloom/HoodieGlobalBloomIndex.java

```diff
@@ -114,14 +117,24 @@ public HoodieGlobalBloomIndex(HoodieWriteConfig config) {
         keyLocationPairRDD.mapToPair(p -> new Tuple2<>(p._1.getRecordKey(), new Tuple2<>(p._2, p._1)));
 
     // Here as the recordRDD might have more data than rowKeyRDD (some rowKeys' fileId is null), so we do left outer join.
-    return incomingRowKeyRecordPairRDD.leftOuterJoin(existingRecordKeyToRecordLocationHoodieKeyMap).values().map(record -> {
+    return incomingRowKeyRecordPairRDD.leftOuterJoin(existingRecordKeyToRecordLocationHoodieKeyMap).values().flatMap(record -> {
       final HoodieRecord<T> hoodieRecord = record._1;
       final Optional<Tuple2<HoodieRecordLocation, HoodieKey>> recordLocationHoodieKeyPair = record._2;
       if (recordLocationHoodieKeyPair.isPresent()) {
         // Record key matched to file
-        return getTaggedRecord(new HoodieRecord<>(recordLocationHoodieKeyPair.get()._2, hoodieRecord.getData()), Option.ofNullable(recordLocationHoodieKeyPair.get()._1));
+        if (config.getGlobalBloomIndexShouldUpdatePartitionPath()
```

Comment:
minor: Would be good if we first check partition path differs and then check config value. Bcoz, the config value doesn't matter if partition path matches. So, for readability purposes would be good to swap the condition checks.

Reply:
I can actually make the argument otherwise :) .. the first check is easier to read and I only need to both with the second one, if first passes.. we can leave this to @xushiyan  to make the call.

Reply:
2nd thought on this: we should treat a configuration as the dominating factor and the partition equality check as a supporting factor. Doing it the other way will diminish the purpose of a configuration. Hence I would keep that order as is. Thanks for bringing thoughts to it!